Automattic/newspack-newsletters
# refactor: ad insertion

### All Submissions:






### Changes proposed in this Pull Request:

Implements support for block count insertion of newsletter ads and refactors the insertion strategy to its own logic outside the renderer class.

The refactored insertion logic is borrowed from [Newspack Popups's inline prompt insertion](https://github.com/Automattic/newspack-popups/blob/f091a24c4cb7e78c114fcb5d473a47c7ef1210aa/includes/class-newspack-popups-inserter.php#L267-L431).

### How to test the changes in this Pull Request:

1. Make sure you have an ESP configured
2. Create a new newsletter ad and make its insertion strategy set to "Block count" and set the count to 3
3. Create another newsletter ad and make its insertion strategy set to "Percentage" and the value to 50%
4. Draft a new newsletter using any layout and click to send
5. Confirm the modal opens with a newsletter preview and the ads are inserted with proper positioning
6. Close the modal and add two "Newsletter Ad" blocks
7. Leave the blocks as "Automatic selection" and position them at the top of the newsletter
8. Save and click "Preview email"
9. Confirm both ads render with the correct order
10. Close the modal and change the ad block to a specific ad
11. Save and click "Preview email"
12. Confirm the expected ad renders

### Other information:
